src/main/java/core/basesyntax/BooleanExpression.java

```diff
@@ -14,6 +14,11 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return (((a ^ b) && (c ^ d))
+                || ((a ^ c) && (b ^ d))
+                || ((a ^ d) && (c ^ b))
+                || ((b ^ c) && (a ^ d))
+                || ((b ^ d) && (c ^ a))
+                || ((c ^ d) && (a ^ b)));
```

Comment:
you've got some redundant check here: 
```suggestion
        return (a ^ b) && (c ^ d) || (a ^ c) && (b ^ d);
```

Reply:
done